lib/travis/build/script/perl6.rb

```diff
@@ -0,0 +1,45 @@
+# Maintained by:
+# Paul Cochrane    @paultcochrane   paul@liekut.de
+
+module Travis
+  module Build
+    class Script
+      class Perl6 < Script
+        DEFAULTS = {
+          perl6: 'latest'
+        }
+
+        def configure
+          super
+
+          sh.echo ''
+          sh.echo 'Perl6 support for Travis-CI is community maintained.', ansi: :red
+          sh.echo 'Please open any issues at https://github.com/travis-ci/travis-ci/issues/new', ansi: :red
+
+          sh.echo 'Installing Rakudo (MoarVM)', ansi: :yellow
+          sh.cmd 'git clone https://github.com/rakudo/rakudo.git'
+          sh.cmd 'cd rakudo'
+          sh.cmd 'sudo perl Configure.pl --backends=moar --gen-nqp --gen-moar --prefix=/usr'
+          sh.cmd 'sudo make install'
```

Comment:
You can see build times listed in the original issue: https://github.com/travis-ci/travis-ci/issues/2978 which use perl.rb. I'd run your code but i've not been able to get local travis testing working.

Also shown in the original issue is the use of 'rakudobrew', a perl program much like perlbrew which allows installation to a custom directory. 
https://github.com/ugexe/zef/blob/master/.travis.yml#L14-L17 for instance builds/installs perl6 and a package manager to ~./rakudobrew/[backend name]/


For an empty module, using the MoarVM backend, it takes ~5-8 minutes to build rakudo perl6, the backend, nqp, and the empty module.

JVM takes significantly longer, but I think getting just MoarVM working first and primarily should be the main focus.